stdlib/public/core/Reverse.swift

```diff
@@ -253,6 +271,16 @@ extension ReversedCollection: BidirectionalCollection {
 
 extension ReversedCollection: RandomAccessCollection where Base: RandomAccessCollection { }
 
+extension ReversedCollection {
+  // This is optimization to return original collection of doubly reversed collection
+  // For example [1,2].reversed().reversed() => [1,2]
+
+  /// Returns the elements of the original collection (reversal of reversal)
+  public func reversed() -> Base {
```

Comment:
Since this is a more concrete overload of an existing method, we should be careful not to break existing code.

The type of `let xs = [1,2,3].reversed().reversed()` changes. This new overload should be `@available(swift, introduced: 4.2)`.